lib/src/test/kotlin/net/dean/jraw/test/unit/GildingsTest.kt

```diff
@@ -0,0 +1,267 @@
+            "\"likes\": null," +
+            "\"suggested_sort\": null," +
+            "\"banned_at_utc\": null," +
+            "\"view_count\": null," +
+            "\"archived\": false," +
+            "\"no_follow\": false," +
+            "\"is_crosspostable\": true," +
+            "\"pinned\": false," +
+            "\"over_18\": false," +
+            "\"media_only\": false," +
+            "\"link_flair_template_id\": null," +
+            "\"can_gild\": true," +
+            "\"spoiler\": false," +
+            "\"locked\": false," +
+            "\"author_flair_text\": null," +
+            "\"visited\": false," +
+            "\"num_reports\": null," +
+            "\"distinguished\": null," +
+            "\"subreddit_id\": \"t5_2qzb6\"," +
+            "\"mod_reason_by\": null," +
+            "\"removal_reason\": null," +
+            "\"link_flair_background_color\": \"\"," +
+            "\"id\": \"a83kug\"," +
+            "\"is_robot_indexable\": true," +
+            "\"report_reasons\": null," +
+            "\"author\": \"_meghanmurphy\"," +
+            "\"num_crossposts\": 5," +
+            "\"num_comments\": 1657," +
+            "\"send_replies\": true," +
+            "\"whitelist_status\": \"all_ads\"," +
+            "\"mod_reports\": []," +
+            "\"author_patreon_flair\": false," +
+            "\"author_flair_text_color\": null," +
+            "\"permalink\": \"/r/IAmA/comments/a83kug/i_am_meghan_murphy_journalist_feminist_and/\"," +
+            "\"parent_whitelist_status\": \"all_ads\"," +
+            "\"stickied\": false," +
+            "\"url\": \"https://www.reddit.com/r/IAmA/comments/a83kug/i_am_meghan_murphy_journalist_feminist_and/\"," +
+            "\"subreddit_subscribers\": 18621904," +
+            "\"created_utc\": 1545346405," +
+            "\"media\": null," +
+            "\"is_video\": false" +
+            "}"
+
+        val adapter = JrawUtils.adapter<Submission>()
+        val submission = adapter.fromJson(json)
+
+        println("[ silver: ${submission?.gildings?.silvers}, gold: ${submission?.gildings?.golds}, platinum: ${submission?.gildings?.platinums} ]")
+
+        it("should have gildings") {
+
+            submission.should.not.be.`null`
+
+            submission?.gildings?.silvers.should.not.be.`null`
+            submission?.gildings?.golds.should.not.be.`null`
+            submission?.gildings?.platinums.should.not.be.`null`
```

Comment:
Once we pass line 123 we know `submission` is non-null. Since Submission.getGildings is `@NotNull`, we can do something like this:

```kotlin
125: submission!!.gildings!!.silvers.should.not.be.`null`
126: submission.gildings.golds.should.not.be.`null`
// etc.
```